lib/source.js

```diff
@@ -47,50 +47,116 @@ module.exports = class CovSource {
 
   // given a start column and end column in absolute offsets within
   // a source file (0 - EOF), returns the relative line column positions.
-  offsetToOriginalRelative (sourceMap, startCol, endCol) {
+  offsetToOriginalRelative (sourceMapScanner, startCol, endCol) {
     const lines = this.lines.filter((line, i) => {
       return startCol <= line.endCol && endCol >= line.startCol
     })
-    if (!lines.length) return {}
-
-    const start = originalPositionTryBoth(
-      sourceMap,
-      lines[0].line,
-      startCol - lines[0].startCol
-    )
-    let end = originalEndPositionFor(
-      sourceMap,
-      lines[lines.length - 1].line,
-      endCol - lines[lines.length - 1].startCol
-    )
-
-    if (!(start && end)) {
-      return {}
+    if (!lines.length) {
+      return []
     }
 
-    if (!(start.source && end.source)) {
-      return {}
-    }
+    const sourceMapIterator = sourceMapScanner.getIterator()
+    let start = sourceMapIterator.scanTo(lines[0].line, startCol - lines[0].startCol)
```

Comment:
If I'm understanding the goal of this PR, it's to make it so that we can handle a source map for a library like `uglify`, which might translate one compiled file into many source files?

It feels like the logic for each individual source file should remain almost the same? I wonder if we could abstract this logic out a layer, such that the top level class has an array of `CovSource` objects produced by some sort of factory.